src/Console/UpgradeCommand.php

```diff
@@ -23,107 +23,72 @@ protected function configure()
-                    'DataList' => 'SilverStripe\Model\DataList',
-                    'DataObject' => 'SilverStripe\Model\DataObject',
-                    'DataObjectInterface' => 'SilverStripe\Model\DataObjectInterface',
-                    'DataQuery' => 'SilverStripe\Model\DataQuery',
-                    'DataQuery_SubGroup' => 'SilverStripe\Model\DataQuery_SubGroup',
-                    'DatabaseAdmin' => 'SilverStripe\Model\DatabaseAdmin',
-                    'GroupedList' => 'SilverStripe\Model\GroupedList',
-                    'SS_HTMLValue' => 'SilverStripe\Model\HTMLValue',
-                    'SS_HTML4Value' => 'SilverStripe\Model\HTML4Value',
-                    'HasManyList' => 'SilverStripe\Model\HasManyList',
-                    'HiddenClass' => 'SilverStripe\Model\HiddenClass',
-                    'Hierarchy' => 'SilverStripe\Model\Hierarchy',
-                    'PolymorphicHasManyList' => 'SilverStripe\Model\PolymorphicHasManyList',
-                    'UnsavedRelationList' => 'SilverStripe\Model\UnsavedRelationList',
-                    'SS_List' => 'SilverStripe\Model\SS_List',
-                    'DataModel' => 'SilverStripe\Model\DataModel',
-                    'Relation' => 'SilverStripe\Model\Relation',
-
-                    'SS_Database' => 'SilverStripe\Model\Connect\Database',
-                    'SilverStripe\Model\Connect\SS_Database' => 'SilverStripe\Model\Connect\Database',
-                    'SS_DatabaseException' => 'SilverStripe\Model\Connect\DatabaseException',
-                    'DBConnector' => 'SilverStripe\Model\Connect\DBConnector',
-                    'DBQueryManager' => 'SilverStripe\Model\Connect\DBQueryManager',
-                    'DBQueryBuilder' => 'SilverStripe\Model\Connect\DBQueryBuilder',
-                    'DBSchemaManager' => 'SilverStripe\Model\Connect\DBSchemaManager',
-                    'MySQLDatabase' => 'SilverStripe\Model\Connect\MySQLDatabase',
-                    'MySQLiConnector' => 'SilverStripe\Model\Connect\MySQLiConnector',
-                    'MySQLQuery' => 'SilverStripe\Model\Connect\MySQLQuery',
-                    'MySQLQueryBuilder' => 'SilverStripe\Model\Connect\MySQLQueryBuilder',
-                    'MySQLSchemaManager' => 'SilverStripe\Model\Connect\MySQLSchemaManager',
-                    'MySQLStatement' => 'SilverStripe\Model\Connect\MySQLStatement',
-                    'PDOConnector' => 'SilverStripe\Model\Connect\PDOConnector',
-                    'PDOQuery' => 'SilverStripe\Model\Connect\PDOQuery',
-                    'SS_Query' => 'SilverStripe\Model\Connect\Query',
-
-                    'SS_Filterable' => 'SilverStripe\Model\Filterable',
-                    'SS_Sortable' => 'SilverStripe\Model\Sortable',
-                    'SS_Limitable' => 'SilverStripe\Model\Limitable',
-                ],
-            ])
+            (new RenameClasses())->withParameters($config)
```

Comment:
For a future PR: the use of `RenameClasses` should be specified in the config file as well as the parameters. This will let config files make use of multiple upgrade rules.